### PR TITLE
doc: add infrastructure for sphinx-generated docs and section 5 and 8 manual pages

### PR DESCRIPTION
**Pull request by grondo**
This is a first draft of adding man pages and html documentation for the flux-security project.

This first step just adds enough to get a single new man page `flux-config-imp(5)` built.

I haven't tested beyond that, the manpage might not get installed yet, etc. However, anyone else that wishes to add some docs could start pushing them here since the basic framework to build them seems to be working.

### CODE REVIEW

**Review by garlick — COMMENTED** [review]
So how does the cool readthedocs preview appear in our CI runs in flux core?  I'm not finding it in the github workflow stuff (possibly I'm missing it)

**Comment by grondo** [comment]
I'm unsure. I think we have to enable readthedocs integration first. I can
try that

File: doc/man5/flux-config-imp.rst [hunk 1]
```diff
@@ -0,0 +1,116 @@
+==================
+flux-config-imp(5)
+==================
+
+
+DESCRIPTION
+===========
+
+The Flux Individual Minister of Privilege (IMP or ``flux-imp``) is a setuid
```
[the remark this answers is not in the document]
Comment (grondo) [reply doc/man5/flux-config-imp.rst:9]: Heh, sorry for the typo. Warning: The man page is a really rough draft right now.

**Comment by grondo** [comment]
Ok, I've made a quick editing pass so that the existing man page isn't completely embarrassing, added a `make check` target to the docs that does a spellcheck, and I think I've got the readthedocs integration working.

Remaining TODOs:

 * Fill in other important missing man pages
 * Ensure manpages are properly installed with `make install` 
 * Figure out how to embed this documentation under our main flux-framework readthedocs site (probably that will be done via flux-docs)

**Comment by grondo** [comment]
Added a very short `flux-imp(8)` man page in case it is a useful reference.

FYI - I didn't document `flux-imp casign` or related configuration. Mostly because I'd have to refresh my memory how it all works before writing any docs.

**Review by garlick — APPROVED** [review]
LGTM!

Since I contributed, IMHO @grondo should review my bit also and add his approval when ready.

**Comment by grondo** [comment]
Thanks @garlick for filling in the majority of the documentation here! I'm going to remove WIP now and give the added files a review.

**Pull request merged**